internal/worker/bootstrap/service.go

```diff
@@ -26,12 +27,23 @@ import (
 
 // ApplicationService instances save an application to dqlite state.
 type ApplicationService interface {
+	// CreateApplication creates a new application with the given name and
+	// charm.
 	CreateApplication(
-		ctx context.Context, name string, charm charm.Charm, origin corecharm.Origin,
-		params applicationservice.AddApplicationArgs, units ...applicationservice.AddUnitArg,
+		context.Context, string, charm.Charm, corecharm.Origin,
+		applicationservice.AddApplicationArgs, ...applicationservice.AddUnitArg,
 	) (coreapplication.ID, error)
 
+	// ResolveControllerCharmDownload resolves the controller charm download slot.
+	ResolveControllerCharmDownload(
+		ctx context.Context,
+		resolve application.ResolveControllerCharmDownload,
+	) (application.ResolvedControllerCharmDownload, error)
```

Comment:
This is unused?

Reply:
It's used in internal/bootstrap package.